source/Halibut.Tests/Timeouts/NextAndProceedControlMessageTimeoutsFixture.cs

```diff
@@ -20,53 +16,47 @@ namespace Halibut.Tests.Timeouts
+                             .AsLatestClientAndLatestServiceBuilder()
+                             .WithPortForwarding(out var portForwarder)
+                             .WithServiceControlMessageObserver(new FuncControlMessageObserver
+                             {
+                                 BeforeSendingControlMessageAction = controlMessage =>
+                                 {
+                                     if (shouldPausePumpOnNextNextControlMessage && controlMessage.Equals("NEXT"))
+                                     {
+                                         Logger.Information("Pausing pump");
+                                         shouldPausePumpOnNextNextControlMessage = false;
+                                         portForwarder.Value!.PauseExistingConnections();
+                                         sw.Start();
+                                     }
+                                 }
+                             })
+                             .As<LatestClientAndLatestServiceBuilder>()
+                             .WithEchoService()
+                             .WithPollingReconnectRetryPolicy(() => new RetryPolicy(1, TimeSpan.Zero, TimeSpan.Zero))
+                             .Build(CancellationToken))
             {
                 var echo = clientAndService.CreateAsyncClient<IEchoService, IAsyncClientEchoService>();
-                
+
                 await echo.SayHelloAsync(Some.RandomAsciiStringOfLength(2000));
-                // --> NEXT sent
-                // --> Proceed sent
-                await Task.Delay(TimeSpan.FromSeconds(3)); // Allow enough time for the polling queue to unnecessarily send a null message down the queue because of a race condition in the queue. 
-                long nextMessageSize = dataSentSizes.Last();
-                pauseStreamWhenServiceSendsMessageOfSize = nextMessageSize; // Lets hope they are all the same size
-                Logger.Information("Will pause the pump next time the polling service sends a message of size {Size}", pauseStreamWhenServiceSendsMessageOfSize);
 
-                await echo.SayHelloAsync(Some.RandomAsciiStringOfLength(2000)); // Second last message
-                // --> NEXT sent and pump paused here.
-                
-                var sw = Stopwatch.StartNew();
-                // Service must detect that it can't get a PROCEED back in time and so must decide to abandon and then try again within
-                // its retry policy.
+                shouldPausePumpOnNextNextControlMessage = true;
+                // When the port forwarder is paused the stopwatch is started,
+                // we time the length it takes for recovery on that paused connection
                 await echo.SayHelloAsync(Some.RandomAsciiStringOfLength(2000));
+                await echo.SayHelloAsync(Some.RandomAsciiStringOfLength(2000)); // Do a second call just in case we missed the NEXT control message
```

Comment:
Why might we miss the NEXT control message?

Reply:
because the polling endpoint comes back and says "NEXT" please before we line up a request for the polling tentacle.